### ISSUE TITLE
Why sending `SEGSEGV` does not terminate the clickhouse-server process

### ISSUE THREAD

**Issue by cangyin** [ClickHouse/ClickHouse#58497]
Sending signal `SIGSEGV` to a running clickhouse-server process just makes it to print `FATAL` logs, and it keeps running.

Send a second or more `SIGSEGV` signals will not get any feed back or `FATAL` logs.

Command used:
```
kill -s SIGSEGV -- $(pidof clickhouse-server >&2; pidof clickhouse-server)
```

Whereas in case of real NULL pointer access segmentation faults (as encountered when modified some code), clickhouse-server will print the fatal log and exit immediately.

**Comment by Felixoid** [comment]
Tested it on version 24.2.2.71. On the sigsegv it produces the next stacktrace, and then finishes all the threads, and exit:

```
2024.03.19 15:19:23.111729 [ 220327 ] {} <Trace> MergedBlockOutputStream: filled checksums 202403_1964_1964_0 (state Temporary)
2024.03.19 15:19:23.112118 [ 220327 ] {} <Trace> system.asynchronous_metric_log (595c7d73-b375-4578-a97c-db7f6d786686): Renaming temporary part tmp_insert_202403_1964_1964_0 to 202403_4921_4921_0 with tid (1, 1, 00000000-0000-0000-0000-000000000000).
2024.03.19 15:19:23.112290 [ 220327 ] {} <Trace> SystemLog (system.asynchronous_metric_log): Flushed system log up to offset 5573676
2024.03.19 15:19:26.914984 [ 219692 ] {} <Trace> BaseDaemon: Received signal 11
2024.03.19 15:19:26.915393 [ 303767 ] {} <Fatal> BaseDaemon: ########## Short fault info ############
2024.03.19 15:19:26.915467 [ 303767 ] {} <Fatal> BaseDaemon: (version 24.2.2.71 (official build), build id: 57F857DCFE8BA6838F6463E4665CD700852BFF0E, git hash: 9293d361e72be9f6ccfd444d504e2137b2e837cf) (from thread 219691) Received signal 11
2024.03.19 15:19:26.915482 [ 303767 ] {} <Fatal> BaseDaemon: Signal description: Segmentation fault
2024.03.19 15:19:26.915494 [ 303767 ] {} <Fatal> BaseDaemon: Address: 0x4a296. Access: read. Unknown si_code.
2024.03.19 15:19:26.915504 [ 303767 ] {} <Fatal> BaseDaemon: Stack trace: 0x000078c04c49febe 0x000078c04c4a2750 0x000000000d2523cb 0x000000000d0d618e 0x00000000158c9606 0x000000000d0c0551 0x00000000158d2479 0x000000000d0bd60a 0x00000000078c8e18 0x000078c04c43dcd0 0x000078c04c43dd8a 0x00000000060c686e
2024.03.19 15:19:26.915518 [ 303767 ] {} <Fatal> BaseDaemon: ########################################
2024.03.19 15:19:26.915529 [ 303767 ] {} <Fatal> BaseDaemon: (version 24.2.2.71 (official build), build id: 57F857DCFE8BA6838F6463E4665CD700852BFF0E, git hash: 9293d361e72be9f6ccfd444d504e2137b2e837cf) (from thread 219691) (no query) Received signal Segmentation fault (11)
2024.03.19 15:19:26.915537 [ 303767 ] {} <Fatal> BaseDaemon: Address: 0x4a296. Access: read. Unknown si_code.
2024.03.19 15:19:26.915544 [ 303767 ] {} <Fatal> BaseDaemon: Stack trace: 0x000078c04c49febe 0x000078c04c4a2750 0x000000000d2523cb 0x000000000d0d618e 0x00000000158c9606 0x000000000d0c0551 0x00000000158d2479 0x000000000d0bd60a 0x00000000078c8e18 0x000078c04c43dcd0 0x000078c04c43dd8a 0x00000000060c686e
2024.03.19 15:19:26.915566 [ 303767 ] {} <Fatal> BaseDaemon: 2. ? @ 0x000078c04c49febe
2024.03.19 15:19:26.915574 [ 303767 ] {} <Fatal> BaseDaemon: 3. ? @ 0x000078c04c4a2750
2024.03.19 15:19:26.915607 [ 303767 ] {} <Fatal> BaseDaemon: 4. ? @ 0x000000000d2523cb
2024.03.19 15:19:26.915622 [ 303767 ] {} <Fatal> BaseDaemon: 5. ? @ 0x000000000d0d618e
2024.03.19 15:19:26.915633 [ 303767 ] {} <Fatal> BaseDaemon: 6. ? @ 0x00000000158c9606
2024.03.19 15:19:26.915645 [ 303767 ] {} <Fatal> BaseDaemon: 7. ? @ 0x000000000d0c0551
2024.03.19 15:19:26.915658 [ 303767 ] {} <Fatal> BaseDaemon: 8. ? @ 0x00000000158d2479
2024.03.19 15:19:26.915670 [ 303767 ] {} <Fatal> BaseDaemon: 9. ? @ 0x000000000d0bd60a
2024.03.19 15:19:26.915682 [ 303767 ] {} <Fatal> BaseDaemon: 10. ? @ 0x00000000078c8e18
2024.03.19 15:19:26.915693 [ 303767 ] {} <Fatal> BaseDaemon: 11. ? @ 0x000078c04c43dcd0
2024.03.19 15:19:26.915705 [ 303767 ] {} <Fatal> BaseDaemon: 12. ? @ 0x000078c04c43dd8a
2024.03.19 15:19:26.915717 [ 303767 ] {} <Fatal> BaseDaemon: 13. ? @ 0x00000000060c686e
2024.03.19 15:19:27.040657 [ 303767 ] {} <Fatal> BaseDaemon: Integrity check of the executable successfully passed (checksum: 3485110FABDB0C94202BD684999A9814)
2024.03.19 15:19:27.040759 [ 303767 ] {} <Debug> SystemLogQueue (system.crash_log): Requested flush up to offset 1
2024.03.19 15:19:27.040795 [ 303767 ] {} <Information> SentryWriter: Not sending crash report
2024.03.19 15:19:27.040795 [ 220372 ] {} <Trace> SystemLog (system.crash_log): Flushing system log, 1 entries to flush up to offset 1
2024.03.19 15:19:27.040806 [ 303767 ] {} <Fatal> BaseDaemon: Report this error to https://github.com/ClickHouse/ClickHouse/issues
2024.03.19 15:19:27.040846 [ 220372 ] {} <Debug> SystemLog (system.crash_log): Creating new table system.crash_log for CrashLog
2024.03.19 15:19:27.057819 [ 220372 ] {} <Debug> system.crash_log (833deb5a-c8ba-4a47-bb66-22e5b1b5454f): Loading data parts
2024.03.19 15:19:27.058010 [ 220372 ] {} <Debug> system.crash_log (833deb5a-c8ba-4a47-bb66-22e5b1b5454f): There are no data parts
2024.03.19 15:19:27.065730 [ 220372 ] {} <Trace> system.crash_log (833deb5a-c8ba-4a47-bb66-22e5b1b5454f): Trying to reserve 1.00 MiB using storage policy from min volume index 0
2024.03.19 15:19:27.065786 [ 220372 ] {} <Trace> DiskLocal: Reserved 1.00 MiB on local disk `default`, having unreserved 956.54 GiB.
2024.03.19 15:19:27.066310 [ 220372 ] {} <Trace> MergedBlockOutputStream: filled checksums all_1_1_0 (state Temporary)
2024.03.19 15:19:27.066696 [ 220372 ] {} <Trace> system.crash_log (833deb5a-c8ba-4a47-bb66-22e5b1b5454f): Renaming temporary part tmp_insert_all_1_1_0 to all_1_1_0 with tid (1, 1, 00000000-0000-0000-0000-000000000000).
2024.03.19 15:19:27.066888 [ 220372 ] {} <Trace> SystemLog (system.crash_log): Flushed system log up to offset 1
2024.03.19 15:19:28.000780 [ 220353 ] {} <Trace> AsynchronousMetrics: MemoryTracking: was 1.18 GiB, peak 15.03 GiB, free memory in arenas 176.00 MiB, will set to 1.07 GiB (RSS), difference: -111.95 MiB
2024.03.19 15:19:28.960185 [ 220370 ] {} <Trace> SystemLog (system.metric_log): Flushing system log, 8 entries to flush up to offset 13765
2024.03.19 15:19:28.973824 [ 220370 ] {} <Trace> system.metric_log (8144ccad-1ab9-4508-aa44-45784d2850f7): Trying to reserve 1.00 MiB using storage policy from min volume index 0
2024.03.19 15:19:28.973880 [ 220370 ] {} <Trace> DiskLocal: Reserved 1.00 MiB on local disk `default`, having unreserved 956.54 GiB.
2024.03.19 15:19:28.980877 [ 220370 ] {} <Trace> MergedBlockOutputStream: filled checksums 202403_1827_1827_0 (state Temporary)
2024.03.19 15:19:28.986422 [ 220370 ] {} <Trace> system.metric_log (8144ccad-1ab9-4508-aa44-45784d2850f7): Renaming temporary part tmp_insert_202403_1827_1827_0 to 202403_4577_4577_0 with tid (1, 1, 00000000-0000-0000-0000-000000000000).
2024.03.19 15:19:28.988175 [ 220370 ] {} <Trace> SystemLog (system.metric_log): Flushed system log up to offset 13765
2024.03.19 15:19:30.112418 [ 220327 ] {} <Trace> SystemLog (system.asynchronous_metric_log): Flushing system log, 2835 entries to flush up to offset 5576511
2024.03.19 15:19:30.115947 [ 220327 ] {} <Trace> system.asynchronous_metric_log (595c7d73-b375-4578-a97c-db7f6d786686): Trying to reserve 1.00 MiB using storage policy from min volume index 0
2024.03.19 15:19:30.115997 [ 220327 ] {} <Trace> DiskLocal: Reserved 1.00 MiB on local disk `default`, having unreserved 956.53 GiB.
2024.03.19 15:19:30.116903 [ 220327 ] {} <Trace> MergedBlockOutputStream: filled checksums 202403_1965_1965_0 (state Temporary)
2024.03.19 15:19:30.117522 [ 220327 ] {} <Trace> system.asynchronous_metric_log (595c7d73-b375-4578-a97c-db7f6d786686): Renaming temporary part tmp_insert_202403_1965_1965_0 to 202403_4922_4922_0 with tid (1, 1, 00000000-0000-0000-0000-000000000000).
2024.03.19 15:19:30.117729 [ 220113 ] {} <Debug> system.asynchronous_metric_log (595c7d73-b375-4578-a97c-db7f6d786686) (MergerMutator): Selected 6 parts from 202403_4902_4917_3 to 202403_4922_4922_0
2024.03.19 15:19:30.117734 [ 220327 ] {} <Trace> SystemLog (system.asynchronous_metric_log): Flushed system log up to offset 5576511
2024.03.19 15:19:30.117757 [ 220113 ] {} <Trace> system.asynchronous_metric_log (595c7d73-b375-4578-a97c-db7f6d786686): Trying to reserve 1.00 MiB using storage policy from min volume index 0
2024.03.19 15:19:30.117770 [ 220113 ] {} <Trace> DiskLocal: Reserved 1.00 MiB on local disk `default`, having unreserved 956.53 GiB.
2024.03.19 15:19:30.117892 [ 220286 ] {595c7d73-b375-4578-a97c-db7f6d786686::202403_4902_4922_4} <Debug> MergeTask::PrepareStage: Merging 6 parts: from 202403_4902_4917_3 to 202403_4922_4922_0 into Compact with storage Full
2024.03.19 15:19:30.117976 [ 220286 ] {595c7d73-b375-4578-a97c-db7f6d786686::202403_4902_4922_4} <Debug> MergeTask::PrepareStage: Selected MergeAlgorithm: Horizontal
2024.03.19 15:19:30.118010 [ 220286 ] {595c7d73-b375-4578-a97c-db7f6d786686::202403_4902_4922_4} <Debug> MergeTreeSequentialSource: Reading 7 marks from part 202403_4902_4917_3, total 45785 rows starting from the beginning of the part
2024.03.19 15:19:30.118061 [ 220286 ] {595c7d73-b375-4578-a97c-db7f6d786686::202403_4902_4922_4} <Debug> MergeTreeSequentialSource: Reading 2 marks from part 202403_4918_4918_0, total 2835 rows starting from the beginning of the part
2024.03.19 15:19:30.118095 [ 220286 ] {595c7d73-b375-4578-a97c-db7f6d786686::202403_4902_4922_4} <Debug> MergeTreeSequentialSource: Reading 2 marks from part 202403_4919_4919_0, total 2835 rows starting from the beginning of the part
2024.03.19 15:19:30.118134 [ 220286 ] {595c7d73-b375-4578-a97c-db7f6d786686::202403_4902_4922_4} <Debug> MergeTreeSequentialSource: Reading 2 marks from part 202403_4920_4920_0, total 2835 rows starting from the beginning of the part
2024.03.19 15:19:30.118166 [ 220286 ] {595c7d73-b375-4578-a97c-db7f6d786686::202403_4902_4922_4} <Debug> MergeTreeSequentialSource: Reading 2 marks from part 202403_4921_4921_0, total 2835 rows starting from the beginning of the part
2024.03.19 15:19:30.118198 [ 220286 ] {595c7d73-b375-4578-a97c-db7f6d786686::202403_4902_4922_4} <Debug> MergeTreeSequentialSource: Reading 2 marks from part 202403_4922_4922_0, total 2835 rows starting from the beginning of the part
2024.03.19 15:19:30.131691 [ 220286 ] {595c7d73-b375-4578-a97c-db7f6d786686::202403_4902_4922_4} <Debug> MergeTask::MergeProjectionsStage: Merge sorted 59960 rows, containing 5 columns (5 merged, 0 gathered) in 0.013852808 sec., 4328364.328733929 rows/sec., 67.02 MiB/sec.
2024.03.19 15:19:30.131997 [ 220286 ] {595c7d73-b375-4578-a97c-db7f6d786686::202403_4902_4922_4} <Trace> MergedBlockOutputStream: filled checksums 202403_4902_4922_4 (state Temporary)
2024.03.19 15:19:30.132551 [ 220286 ] {595c7d73-b375-4578-a97c-db7f6d786686::202403_4902_4922_4} <Trace> system.asynchronous_metric_log (595c7d73-b375-4578-a97c-db7f6d786686): Renaming temporary part tmp_merge_202403_4902_4922_4 to 202403_4902_4922_4 with tid (1, 1, 00000000-0000-0000-0000-000000000000).
2024.03.19 15:19:30.132623 [ 220286 ] {595c7d73-b375-4578-a97c-db7f6d786686::202403_4902_4922_4} <Trace> system.asynchronous_metric_log (595c7d73-b375-4578-a97c-db7f6d786686) (MergerMutator): Merged 6 parts: [202403_4902_4917_3, 202403_4922_4922_0] -> 202403_4902_4922_4
2024.03.19 15:19:30.132752 [ 220286 ] {} <Debug> MemoryTracker: Peak memory usage background process to apply mutate/merge in table: 7.45 MiB.
2024.03.19 15:19:33.081410 [ 219690 ] {} <Fatal> Application: Child process was terminated by signal 11.
```

**Comment by alexey-milovidov** [comment]
ClickHouse has a signal handler to print these diagnostics.
